src/Lens/Micro/Aeson.hs

```diff
@@ -355,7 +334,9 @@ members = _Object . traverse
 -- >>> "[1,2,3]" & nth 1 .~ Number 20
 -- "[1,20,3]"
 nth :: AsValue t => Int -> Traversal' t Value
-nth i = _Array . ix i
+nth i = _Array . \f a -> if 0 <= i && i < V.length a
+  then f (a V.! i) <&> \v -> a V.// [(i, v)]
+  else pure a
```

Comment:
Please turn the lambda into a named function within `where` clause and use guards instead of if-then-else.